# Chapter 4 The Standard Deviation and Other Measures of Dispersion

## Dispersion, or variation
The degree to which numerical data tend to spread about an average value is called the _dispersion_, or _variation_, of the data. Various measures of this dispersion (or variation) are available, the most common being the range, mean deviation, semi-interquartile reange, $10$&mdash;$90$ percentile range, and standard deviation.

---

## The Range
The _range_ of a set of numbers is the difference between the largest and smallest numbers in the set.

---

__EXAMPLE 1.__ The range of the set $2, 3, 3, 5, 5, 5, 8, 10, 12$ is $12-2=10$. Sometimes the range is given by simply quoting the smallest and largest numbers; in the above set, for instance, the range could be indicated as $2$ to $12$, or $2-12$.

---

__4.1.__ Find the range of the sets (a) 12, 6, 7, 3, 15, 10, 18, 5 and (b) 9, 3, 8, 8, 9, 8, 9, 18

### Solution
In both cases, range $=$ largest number $-$ smallest number $=18-3=15$. However, as seen from the arrays of sets (a) and (b),<p>
$$
\begin{align*}
\text{(a)}\ 3, 5, 6, 7, 10, 12, 15, 18\ \ \ \ \ \ \ \ \text{(b)}\ 3, 8, 8, 9, 9, 9, 18
\end{align*}
$$<p>
there is much more variation or dispersion, in (a) than in (b). In fact, (b) consists mainly of $8$'s and $9$'s.<p>
Since the range indicates no difference between the sets, it is not a very good measure of dpsersion in this case. Where extreme values are present, the range is generally a poor measure of dispersion.<p>
An improvement is achieved by throwing out the extreme cases, $3$, and $18$. Then for set (a) the range is $(15 - 5) = 10$, while for set (b) the range is $(9-8)=1$, clearly showing that (a) has greater dispersion than (b). However, this is not the way the range is defined. The semi-interquartile range and the 10&mdash;90 percentile range were designed to improve on the range by eliminating extreme cases.

In [1]:
p1.a <- c(12, 6, 7, 3, 15, 10, 18, 5)
p1.b <- c(9, 3, 8, 8, 9, 8, 9, 18)
(p1.a.range <- diff(range(p1.a)))
(p1.b.range <- diff(range(p1.b)))

[1] 15

[1] 15

---

__4.2.__ Find the range of heights of the students at XYZ University. 

In [2]:
load("Saves/Table_2.1.Rdata")
Table_2.1

60-62,5
63-65,18
66-68,42
69-71,27
72-74,8


### Solution
There are two ways of defining the range for grouped data.
#### First method
$$
\begin{align*}
\text{Range}=\text{class mark of highest class}-\text{class mark of lowest class}=73-61=12\text{in}
\end{align*}
$$
#### Second method
$$
\begin{align*}
\text{Range}=\text{upper class boundary of highest class}-\text{lower class boundary of lowest class}=74.5-59.5=15\text{in}
\end{align*}
$$<p>
The first method tends to eliminate extreme cases to some extent.

In [3]:
# First method - use class marks
# This method tends to eliminate extreme cases to some extent
p2.marks <- c(61, 64, 67, 70, 73)
(p2.xyz.mark.range <- diff(range(p2.marks)))

[1] 12

In [4]:
# Second method - use class boundaries
p2.boundaries <- c(59.5, 62.5, 65.5, 68.5, 71.5, 74.5)
(p2.xyz.boundary.range <- diff(range(p2.boundaries)))

[1] 15

---

## The mean deviation
The _mean deviation_, or _average deviation_, of a set of $N$ numbers $X_1, X_2,\ldots,X_N$ is abbreviaged $\text{MD}$ and is defined.<p>
\begin{align}
\text{MD}
= \frac{\sum^N_{j=1}{\lvert X_j-\bar{X}\rvert}}{N}
= \frac{\sum{\lvert X-\bar{X}\rvert}}{N}
= \frac{\sum{|X - \bar{X}|}}{N}
= \overline{\lvert X-\bar{X}\rvert}\label{eq1}\tag{1}
\end{align}<P>
where $\bar{X}$ is the arithmetic mean of the numbers and $\lvert X_j-\bar{X}\rvert$ is the absolute value of the deviation of $X_j$ from $\bar{X}$. (The _absolute value_ of a number is the number without the associated sign and is indicated by two vertical lines placed around the number; thus $\lvert-4\rvert=4,\lvert+3\rvert=3,\lvert6\rvert=6,$ and $\vert-0.84\rvert=0.84$.)<p>
__EXAMPLE 2.__  Find the mean deviation of the set $2, 3, 6, 8, 11$.<p>
\begin{align}
\text{Arithmetic mean (}\bar{X}\text{)}=\frac{2+3+6+8+11}{5}=6
\end{align}<p>
\begin{align}
\text{MD}
=\frac{\lvert2-6\rvert+\lvert3-6\rvert+\lvert6-6\rvert+\lvert8-6\rvert+\lvert11-6\rvert}{5}
=\frac{\lvert-4\rvert+\lvert-3\rvert+\lvert0\rvert+\lvert2\rvert+\lvert5\rvert}{5}
=\frac{4+3+0+2+5}{5}=2.8
\end{align}<p>
If $X_1, X_2,\ldots,X_K$ occur with frequencies $f_1,f_2,\ldots,f_K$, respectively, the mean deviation can be written as<p>
\begin{align}
\text{MD}
=\frac{\sum^K_{j=1}{f_j\lvert X_j-\bar{X}\rvert}}{N}
=\frac{\sum{f\lvert X-\bar{X}\rvert}}{N}
=\overline{\lvert X-\bar{X}\rvert}\label{eq2}\tag{2}
\end{align}<p>
where $N=\sum^K_{j=1}{f_j}=\sum{f}$. This form is useful for grouped data, where the $X_j$'s represent class marks and the $f_j$'s are the corresponding class frequencies.<p>
Occasionally the mean deviation is defined in terms of absolute deviations from the median or other average instead of from the mean. An interesting property of the sum $\sum^N_{j=1}\lvert X_j-a\rvert$ is that it is a minimum where $a$ is the median (i.e., the mean deviation about the median is a minimum).<p>
Note that it would be more appropriate to use the terminology _mean absolute deviation_ than _mean deviation_.

Below is a function that can be used to find the mean deviation. It is also found in the saved data file `Saves/mean_deviation.Rdata`.

In [5]:
md <- function(x) {
    return (sum(abs(x - mean(x)))/length(x))
}

---

__4.3.__ Find the mean deviation of the sets of numbers in problem 1.

### Solution
(a) The arithmetic mean is<p>
$$
\begin{align*}
\bar{X}=\frac{12+6+7+3+15+10_18+5}{8}=\frac{76}{8}=9.5
\end{align*}
$$
<p>
The mean deviation is<p>
$$
\begin{align*}
\text{MD}
=\frac{\lvert12-9.5\rvert+\lvert6-9.5\rvert+\lvert3-9.5\rvert+\lvert15-9.5\rvert+\lvert10-9.5\rvert+\lvert18-9.5\rvert+\lvert5-9.5\rvert}{8}
=\frac{2.5+3.5+2.5+6.5+5.5+0.5+8.5+4.5}{8}
=\frac{34}{8}=4.25
\end{align*}
$$
<p>
(b)
$$
\begin{align*}
\bar{X}=\frac{9+3+8+8+9+8+9+18}{8}=\frac{72}{8}=9
\end{align*}
$$
<p>
$$
\begin{align*}
\text{MD}
=\frac{\lvert9-9\rvert+\lvert3-9\rvert+\lvert8-9\rvert+\lvert8-9\rvert+\lvert9-9\rvert+\lvert8-9\rvert+\lvert9-9\rvert+\lvert18-9\rvert}{8}
=\frac{0+6+1+1+0+1+0+9}{8}=2.25
\end{align*}
$$
<p>
The mean deviation indicates that set (b) shows less dispersion than set (a), as it should.

In [6]:
p3.x <- c(12, 6, 7, 3, 15, 10, 18, 5)
p3.y <- c(9, 3, 8, 8, 9, 8, 9, 18)
(p4.MD.x <- md(p3.x))
(p4.MD.y <- md(p3.y))

[1] 4.25

[1] 2.25

---

__4.4.__ Find the mean deviation of the heights of the 100 male students at XYZ University.

In [7]:
Table_2.1

60-62,5
63-65,18
66-68,42
69-71,27
72-74,8


### Solution
From Problem 3.20, $\bar{X}=67.45$in. The work can be arranged as in This table:

In [8]:
p4.table <- matrix(c(61, 64, 67, 70, 73, 6.45, 3.45, 0.45, 2.55, 5.55, 5, 18, 42, 27, 8, 32.25, 62.10, 18.90, 68.85, 44.40), nrow=5)
rownames(p4.table) <- c("60-62", "63-65", "66-68", "69-71", "72-74")
colnames(p4.table) <- c("Class Mark (X)", "|X - \\bar{X}| = |X - 67.45|","Frequency (f)", "f|X - \\bar{X}|")
p4.table

,Class Mark (X),|X - \bar{X}| = |X - 67.45|,Frequency (f),f|X - \bar{X}|
60-62,61,6.45,5,32.25
63-65,64,3.45,18,62.10
66-68,67,0.45,42,18.90
69-71,70,2.55,27,68.85
72-74,73,5.55,8,44.40


$$
\begin{align*}
\text{MD}=\frac{\sum{f\lvert X-\bar{X}\rvert}}{N}=\frac{226.50}{100}=2.26\text{in}
\end{align*}
$$
<p>

In [9]:
p4.f <- number.of.students <- as.vector(Table_2.1)
p4.N <- sum(p4.f)
p4.X <- marks <- c(61, 64, 67, 70, 73)
p4.A <- guess <- 67
p4.d <- p4.X - p4.A
p4.X.bar <- p4.avg <- p4.A + sum(p4.f*p4.d)/p4.N
p4.dev <- abs(p4.X - p4.X.bar)
p4.fX <- p4.f*p4.dev
p4.table <- matrix(c(p4.X, p4.dev, p4.f, p4.fX), nrow=length(p4.X))
rownames(p4.table) <- rownames(Table_2.1)
colnames(p4.table) <- c("X", "|X-\\bar{X}|", "f", "f|X-\\bar{X}|")
p4.fX.sum <- sum(p4.fX)
(p4.MD.xyz <- signif(p4.fX.sum/p4.N, 4))
p4.MD.xyz <- 2.26
p4.boundaries <- c(59.5, 62.5, 65.5, 68.5, 71.5, 74.5)

[1] 2.265

---

__4.5.__ Determine the percentage of the students' heights in Problem 4 that fall within the ranges

(a) $\bar{X}\pm\text{MD}$

### Solution
The range from $65.29$ to $69.71$ in is $\bar{X}\pm\text{MD}=67.45\pm2.26$. This range includes all individuals in the third class $+\frac{1}{3}(65.5-65.19)$ of the students in the second class $+\frac{1}{3}(69.71-68.5)$ of the students in the fourth class (since the class-interval size is $3$in, the upper class boundary of the second class is $65.5$in, and the lower class boundary of the fourth class is $68.5$in). The number of students in the range $\bar{X}\pm\text{MD}$ is
<p>
$$
\begin{align*}
42+\frac{0.31}{3}(18)+\frac{1.21}{3}(27)=42+1.86+10.89=54.75\ \ \ \ \text{or}\ \ \ \ 55
\end{align*}
$$
<p>
    which is $55\%$ of the total.

In [10]:
p5.r1 <- round(range(p4.X.bar-p4.MD.xyz, p4.X.bar+p4.MD.xyz),2)
(round(p5.r1.pct <- 42 + (p4.boundaries[3]-p5.r1[1])/3*18 + (p5.r1[2]-p4.boundaries[4])/3*27,0))

[1] 55

(b) $\bar{X}\pm2\text{MD}$

### Solution
The range from $62.93$ to $71.97$ in is $\bar{X}\pm2\text{MD}=67.45\pm2(2.26)=67.45\pm4.52$. The number of students in the range $\bar{X}\pm2\text{MD}$ is
<p>
$$
\begin{align*}
18-\left(\frac{62.93-62.5}{3}\right)(18)+42+27+\left(\frac{71.97-71.5}{3}\right)(8)=85.67\ \ \ \ \text{or}\ \ \ \ 86
\end{align*}
$$
<p>
which is $86\%$

In [11]:
p5.r2 <- round(range(p4.X.bar-2*p4.MD.xyz, p4.X.bar+2*p4.MD.xyz),2)
(round(p5.r2.pct <- 18 - (p5.r2[1]-p4.boundaries[2])/3*18 + 42 + 27 + (p5.r2[2]-p4.boundaries[5])/3*8,0))

[1] 86

and (c) $\bar{X}\pm3\text{MD}$

### Solution
The range from $60.67$ to $74.23$ in is $\bar{X}\pm3\text{MD}=67.45\pm3(2.26)=67.45\pm6.78$. The number of students int he range $\bar{X}\pm3\text{MD}$ is<p>
$$
\begin{align*}
5-\left(\frac{60.67-59.5}{3}\right)(5)+18+42+27+\left(\frac{74.5-74.23}{3}\right)(8)=97.33\ \ \ \ \text{or}\ \ \ \ 97
\end{align*}
$$
    which is $97\%$ of the total.

In [12]:
p5.r3 <- round(range(p4.X.bar-3*p4.MD.xyz, p4.X.bar+3*p4.MD.xyz),2)
(round(p5.r3.pct <- 5 - (p5.r3[1] - p4.boundaries[1])/3*5 + 18 + 42 + 27 + (p4.boundaries[6] - p5.r3[2])/3*8,0))

[1] 91

---

## The Semi-Interquartile Range
The _semi-interquartile range_, or _quartile deviation_, of a set of data is denoted by $Q$ and is defined by<p>
\begin{align}
Q = \frac{Q_3 - Q_1}{2}\label{eq3}\tag{3}
\end{align}<p>
where $Q_1$ and $Q_3$ are the first and third quartiles for the data (see Problems 4.6 and 4.7). The interquartile range $Q_3-Q_1$ is sometimes used, but the semi-interquartile range is more common as a measure of dispersion.

---

__4.6.__ Find the semi-interquartile range for the height distribution of the students at XYZ University.

In [13]:
Table_2.1

60-62,5
63-65,18
66-68,42
69-71,27
72-74,8


### Solution
The lower and upper quartiles are $Q_1=65.5+\frac{2}{42}(3)=65.64$in and $Q_3=68.5+\frac{10}{27}(3)=69.61$in, respectively, and the semi-interquartile range (or quartile deviation) is $Q=\frac{1}{2}(Q_3-Q_1)=\frac{1}{2}(69.61-65.64)=1.98$in. Note that $50\%$ of the cases lie between $Q_1$ and $Q_3$ (i.e., $50$ students have heights between $65.64$ and $69.61$in).<p>
We can consider $\frac{1}{2}(Q_1+Q_3)=67.63$in to be a measure of central tendency (i.e., average height). It follwos that $50\%$ of the heights lie in the range $67.63\pm1.98$in.

In [14]:
p6.ci <- class.interval <- 3
p6.boundaries <- seq(59.5, 74.5, by=class.interval)
p6.cumsum <- cumsum(as.vector(Table_2.1))
p6.f.sum <- sum(Table_2.1)
p6.Q.breaks <- p6.f.sum/4

p6.Q1.index <- match(Table_2.1[p6.cumsum > p6.Q.breaks][1],Table_2.1)
p6.Q1.close.class <- p6.boundaries[p6.Q1.index]
p6.Q1.class <- as.vector(Table_2.1)[p6.Q1.index]
p6.Q1.remainder <- p6.Q.breaks - p6.cumsum[p6.Q1.index-1]
p6.Q1 <- round(p6.Q1.close.class + p6.Q1.remainder/p6.Q1.class*p6.ci,2)

p6.Q3.index <- match(Table_2.1[p6.cumsum > 3*p6.Q.breaks][1], Table_2.1)
p6.Q3.close.class <- p6.boundaries[p6.Q3.index]
p6.Q3.class <- as.vector(Table_2.1)[p6.Q3.index]
p6.Q3.remainder <- 3*p6.Q.breaks - p6.cumsum[p6.Q3.index-1]
p6.Q3 <- round(p6.Q3.close.class + p6.Q3.remainder/p6.Q3.class*p6.ci,2)

c((p6.Q.central <- round((p6.Q1 + p6.Q3)/2,2)), (p6.Q.pm <- round((p6.Q3 - p6.Q1)/2,2)))
(p6.Q.range <- c(p6.Q.central-p6.Q.pm, p6.Q.central+p6.Q.pm))

[1] 67.62  1.98

[1] 65.64 69.60

---

__4.7.__ Find the semi-interquartile range for the wages of the $65$ employees at the P&R Company in Table 2.6.

In [15]:
load("Saves/Table_2.6.Rdata")
Table_2.6

$250.00-$259.99,8
260.00-269.99,10
270.00-279.99,16
280.00-289.99,14
290.00-299.99,10
300.00-309.99,5
310.00-319.99,2


### Solution
From Problem 3.44, $Q_1=\$268.25$ and $Q_3=\$290.75$. Thus the semi-interquartile range $Q=\frac{1}{2}(Q_3-Q_1)=\frac{1}{2}(\$290.75-\$268.25)=\$11.25$. Since $\frac{1}{2}(Q_1+Q_3)=\$279.50$, we can conclude that $50\%$ of the employees earn wages lying in the range $\$279.50\pm\$11.25$.

In [16]:
p7.ci <- class.interval <- 10
p7.boundaries <- seq(249.995, 319.995, by=class.interval)
p7.cumsum <- cumsum(as.vector(Table_2.6))
p7.f.sum <- sum(Table_2.6)
p7.Q.breaks <- p7.f.sum/4

p7.Q1.index <- match(TRUE, p7.cumsum > p7.Q.breaks)[1]
p7.Q1.close.class <- p7.boundaries[p7.Q1.index]
p7.Q1.class <- as.vector(Table_2.6)[p7.Q1.index]
p7.Q1.remainder <- p7.Q.breaks - p7.cumsum[p7.Q1.index-1]
p7.Q1 <- round(p7.Q1.close.class + p7.Q1.remainder/p7.Q1.class*p7.ci,2)

p7.Q3.index <- match(TRUE, p7.cumsum > 3*p7.Q.breaks)[1]
p7.Q3.close.class <- p7.boundaries[p7.Q3.index]
p7.Q3.class <- as.vector(Table_2.6)[p7.Q3.index]
p7.Q3.remainder <- 3*p7.Q.breaks - p7.cumsum[p7.Q3.index-1]
p7.Q3 <- round(p7.Q3.close.class + p7.Q3.remainder/p7.Q3.class*p7.ci,2)

c((p7.Q.central <- round((p7.Q1 + p7.Q3)/2,2)), (p7.Q.pm <- round((p7.Q3 - p7.Q1)/2,2)))
(p7.Q.range <- c(p7.Q.central - p7.Q.pm, p7.Q.central + p7.Q.pm))

[1] 279.49  11.25

[1] 268.24 290.74

---

## The 10-90 Percentile Range
The $10$&mdash;$90$ _percentile range_ of a set of data is defined by<p>
\begin{align}
10-90\ \text{percentile range}=P_{90}-P_{10}\label{eq4}\tag{4}
\end{align}<p>
where $P_{10}$ and $P_{90}$ are the $10$th and $90$th percentiles for the data (see Problem 4.8). The semi-$10$&mdash;$90$ percentile range, $\frac{1}{2}(P_{90}-P_{10})$, can also be used but is not commonly employed.

---

__4.8.__ Find the $10$&mdash;$90$ percentile range of the heights of the students at XYZ University (Table 2.1).

In [17]:
Table_2.1

60-62,5
63-65,18
66-68,42
69-71,27
72-74,8


### Solution
Here $P_{10}=62.5+\frac{5}{18}(3)=63.33$in, and $P_{90}=68.5+\frac{25}{27}(3)=71.27in$. Thus the $10$&mdash;$90$ percentile range is $P_{90}-P_{10}=71.27-63.33=7.94$in. Since \frac{1}{2}(P_{10}+P_{90})=67.30$in and \frac{1}{2}(P_{90}-P_{10})=3.97$in, we can conclude that $80\%$ of the students have heights in the range $67.30\pm3.97$in.

In [18]:
p8.ci <- 3
p8.boundaries <- seq(59.5, 74.5, by=3)
p8.D.cumsum <- cumsum(as.vector(Table_2.1))
p8.f.sum <- sum(Table_2.1)
p8.D.breaks <- p8.f.sum/10

p8.D.10.index <- match(TRUE, p8.D.cumsum > p8.D.breaks)[1]
p8.D.10.close.class <- p8.boundaries[p8.D.10.index]
p8.D.10.class <- as.vector(Table_2.1)[p8.D.10.index]
p8.D.10.remainder <- p8.D.breaks - p8.D.cumsum[p8.D.10.index-1]
p8.D.10 <- round(p8.D.10.close.class + p8.D.10.remainder/p8.D.10.class*p8.ci,2)

p8.D.90.index <- match(TRUE, p8.D.cumsum > 9*p8.D.breaks)[1]
p8.D.90.close.class <- p8.boundaries[p8.D.90.index]
p8.D.90.class <- as.vector(Table_2.1)[p8.D.90.index]
p8.D.90.remainder <- 9*p8.D.breaks - p8.D.cumsum[p8.D.90.index-1]
p8.D.90 <- round(p8.D.90.close.class + p8.D.90.remainder/p8.D.90.class*p8.ci,2)

p8.D.90 - p8.D.10
c((p8.D.central <- round((p8.D.10 + p8.D.90)/2,2)), (p8.D.pm <- round((p8.D.90 - p8.D.10)/2,2)))
(p8.D.10.90.range <- c(p8.D.central - p8.D.pm, p8.D.central + p8.D.pm))

[1] 7.95

[1] 67.31  3.98

[1] 63.33 71.29

---

## The Standard Deviation
The _standard deviation_ of a set of $N$ numbers $X_1,X_2,\ldots,X_N$ is denoted by $s$ and is defined by<p>
\begin{align}
s
= \sqrt{\frac{\sum^N_{j=1}{(X_j - \bar{X})^2}}{N}}
= \sqrt{\frac{\sum{(X=\bar{X})^2}}{N}}
= \sqrt{\frac{\sum{x^2}}{N}}
= \sqrt{(X - \bar{X})^2}\label{eq5}\tag{5}
\end{align}<p>
where $x$ represents the deviations of each of the numbers $X_j$ from the mean $\bar{X}$. Thus $s$ is the root mean square (RMS) of the deviations from the mean, or, as it is sometimes called, the _root-mean-square deviation_.<p>
If $X_1,X_2,\ldots,X_K$ occur with frequencies $f_1,f_2,\ldots,f_k$, respectively, the standard deviation can be written<p>
\begin{align}
s
=\sqrt{\frac{\sum^K_{j=1}{f_j(X_j-\bar{X})^2}}{N}}
=\sqrt{\frac{\sum{f(X-\bar{X})^2}}{N}}
=\sqrt{\frac{\sum{fX^2}}{N}}
=\sqrt{\overline{(X-\bar{X})^2}}\label{eq6}\tag{6}
\end{align}<p>
where $N=\sum^K_{j=1}{f_j}=\sum{f}$. In this form it is useful for grouped data.<p>
Sometimes the standard deviation of a sample's data is defined with $(N-1)$ replacing $N$ in the denominators of the expressions in equations (\ref{eq5}) and (\ref{eq6}) because the resulting value represents a better estimate of the standard deviation of a populiation from which the sample is taken. For large values of $N$ (certainly $N>30$), there is practically no difference between the two definitions. Also, when the better estimate is needed we can always obtain it by multiplying the standard deviation computed according to the first definition by $\sqrt{\frac{N}{(N-1)}}$. Hence we shall adhere to the form (5) and (6).

In [19]:
sd <- function(x) {
        return (sqrt(sum((x - mean(x))^2)/length(x)))
}

---

__4.9.__ Find the standard deviation $s$ of each set of numbers in Problem 1.

In [20]:
(p9.a.s <- round(sd(p1.a),2))
(p9.b.s <- round(sd(p1.b),2))

[1] 4.87

[1] 3.87

---

__4.10.__ The standard deviations of the two data sets given in Problem 4.1 were found using MINITAB and the results are shown below. Compare the answers with those obtained in Problem 4.9.<p>
```
MTB > print c1
set1
      12   6    7    3    15   10  18    5
MTB > print c2
set2
      9    3    8    8    9    8    9    18
MTB > standard deviation c1
```
Column Standard Deviation
```
    Standard deviation of set1 = 5.21
MTB > standard deviation c2
```
Column Standard Deviation
```
    Standard deviation of set2 = 4.14
```

### Solution
The MINITAB package uses the formula
$$
\begin{align*}
s = \sqrt{\frac{\sum{X-\bar{X})^2}}{N-1}}
\end{align*}
$$<p>
and therefore the standard deviations are not the same in Problem 4.9 and 4.10. The answers in Problem 4.10 are obtainable from those in problem 4.9 if we multiply those in Problem 4.9 by $\sqrt{\frac{N}{(N-1)}}$. Since $N=8$ for both sets $\sqrt{\frac{N}{(N-1)}}=1.069045$, and for set $1$, we have $(1.069045)(4.87)=5.21$, the standard deviation given by MINITAB. Similarly, $(1.069045)(3.87)=4.14$, the standard deviation given for set $2$ by MINITAB.

__When should the SD be computed with a denominator of n instead of (n-1)?__

Statistics books often show two equations to compute the SD, one using n, and the other using n-1, in the denominator. Some calculators have two buttons. 

The n-1 equation is used in the common situation where you are analyzing a sample of data and wish to make more general conclusions. The SD computed this way (with n-1 in the denominator) is your best guess for the value of the SD in the overall population.

If you simply want to quantify the variation in a particular set of data, and don't plan to extrapolate to make wider conclusions, then you can compute the SD using n in the denominator. The resulting SD is the SD of those particular values. It makes no sense to compute the SD this way if you want to estimate the SD of the population from which those points were drawn. It only makes sense to use n in the denominator when there is no sampling from a population, there is no desire to make general conclusions. 

The goal of science is always to generalize, so the equation with n in the denominator should not be used. The only example I can think of where it might make sense is in quantifying the variation among exam scores. But much better would be to show a scatterplot of every score, or a frequency distribution histogram. 

GraphPad Prism and InStat always compute the SD with the n-1 denominator.

---

__4.11.__ Find the standard deviation of the heights of the 100 male students at XYZ University (Table 2.1).

### Solution
From Problem 3.15, 3.20, or 3.22, $\bar{X}=67.45$in. The work can be arranged as in Table 4.2.<p>
\begin{align}
s
=\sqrt{\frac{\sum{f(X-\bar{X})^2}}{N}}
=\sqrt{\frac{852.7500}{100}}
=\sqrt{8.5275}=2.92\text{in}
\end{align}

In [21]:
p11.f <- as.vector(Table_2.1)
p11.X <- c(61, 64, 67, 70,73)
p11.X.bar <- round(sum(p11.f*p11.X)/sum(p11.f),2)
p11.X.minus.X.bar <- round((p11.X - p11.X.bar),2)
p11.X.minus.X.bar.squared <- p11.X.minus.X.bar^2
p11.f.X.minus.X.bar.squared <- p11.f*p11.X.minus.X.bar.squared
p11.table <- matrix(c(p11.X, p11.X.minus.X.bar, p11.X.minus.X.bar.squared, p11.f, p11.f.X.minus.X.bar.squared), nrow=length(p11.f))
rownames(p11.table) <- rownames(Table_2.1)
colnames(p11.table) <- c("Class Mark (X)", "X - \\bar{X} = X - 67.45", "(X - \\bar{X})^2", "Frequency (f)", "f(X - \bar{X})^2")
p11.table

,Class Mark (X),X - \bar{X} = X - 67.45,(X - \bar{X})^2,Frequency (f),f(X - ar{X})^2
60-62,61,-6.45,41.6025,5,208.0125
63-65,64,-3.45,11.9025,18,214.2450
66-68,67,-0.45,0.2025,42,8.5050
69-71,70,2.55,6.5025,27,175.5675
72-74,73,5.55,30.8025,8,246.4200


In [22]:
(p11.sd <- round(sqrt(sum(p11.f*(p11.X-p11.X.bar)^2)/sum(p11.f)),2))

[1] 2.92

---

## The variance
The _variance_ of a set of data is defined as the square of the standard deviation and is thus given by $s^2$ in equations (5) and (6).<p>
When it is necessary to distinguish the standard deviation of a population from the standard deviation of a sample drawn from this population, we often use the symbol $s$ for the latter and $\sigma$ (lowercase Greek _sigma_) for the former. Thus $s^2$ and $\sigma^2$ would represent the _sample variance_ and _population variance_, respectively.

---

## Short methods for computing the standard deviation
Equations (5) and (6) can be written, respectively, in the equivalent forms
\begin{align}
s
=\sqrt{\frac{\sum^N_{j=1}{X_j^2}}{N}-\left(\frac{\sum^N_{j=1}{X_j}}{N}\right)^2}
=\sqrt{\frac{\sum{X^2}}{N}-\left(\frac{\sum{X}}{N}\right)^2}
=\sqrt{\overline{X^2}-\bar{X}^2}\label{eq7}\tag{7}
\end{align}<p>
\begin{align}
s
=\sqrt{\frac{\sum^K_{j=1}{f_jX_j^2}}{N}-\left(\frac{\sum^N_{j=1}{X_j}}{N}\right)^2}
=\sqrt{\frac{\sum{fX^2}}{N}-\left(\frac{\sum{fX}}{N}\right)^2}
=\sqrt{\overline{X^2}-\bar{X}^2}\label{eq8}\tag{8}
\end{align}<p>
where $\overline{X^2}$ denotes the mean of the squares of the various values of $X$, while $\bar{X}^2$ denotes the square of the mean of the various values of X (see Problems 4.12 to 4.14).<p>
If $d_j=X_j-A$ are the deviations of $X_j$ from some arbitrary constant A, results (\ref{eq7}) and (\ref{eq8}) become, resqpectively,<p>
\begin{align} 
s
=\sqrt{\frac{\sum^N_{j=1}{d_j^2}}{N}-\left(\frac{\sum^N_{j=1}{d_j}}{N}\right)^2}
=\sqrt{\frac{\sum{d^2}}{N}-\left(\frac{\sum{d}}{N}\right)^2}
=\sqrt{\overline{d^2}-\bar{d}^2}\label{eq9}\tag{9}
\end{align}<p>
\begin{align} 
s
=\sqrt{\frac{\sum^N_{j=1}{f_jd_j^2}}{N}-\left(\frac{\sum^N_{j=1}{f_jX_j}}{N}\right)^2}
=\sqrt{\frac{\sum{fd^2}}{N}-\left(\frac{\sum{fd}}{N}\right)^2}
=\sqrt{\overline{d^2}-\bar{d}^2}\label{eq10}\tag{10}
\end{align}<p>
(See Problems 4.15 and 4.17.)<p>
When data is grouped into a frequency distribution whose class intervals have equal size $c$, we have $d_j=cu_j$ or $X_j=A+cu_j$ and result (10) becomes<p>
\begin{align}
s
=c\sqrt{\frac{\sum^K_{j=1}{f_ju_j^2}}{N}-\left(\frac{\sum^K_{j=1}{f_ju_j}}{N}\right)^2}
=c\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
=c\sqrt{\overline{u^2}-\bar{u}^2}\label{eq11}\tag{11}
\end{align}<p>
This last formula provides a very short method for computing the standard deviation and should always be used for grouped data when the class-interval sizes are equal. It is called the _coding method_ and is exactly analogous to that used in Chapter 3 for computing the arithmetic mean of grouped data. (See Problems 4.16 to 4.19.)


---

## Computing the standard deviations from grouped data

__4.12.__ (a) Prove that
\begin{align}
s
= \sqrt{\frac{\sum{X^2}}{N}-\left(\frac{\sum{X}}{N}\right)^2}
= \sqrt{\overline{X^2}-\bar{X}^2}
\end{align}<p>
(b) Use the formula in part (a) to find the standard deviation of the set $12, 6, 7, 3, 15, 10, 18, 5$.

### Solution
(a) By definition,
\begin{align}
s
=\sqrt{\frac{\sum{(X-\bar{X})^2}}{N}}
\end{align}<p>
Then
\begin{align}
s^2
=\frac{\sum{(X-\bar{X})^2}}{N}
=\frac{\sum{(X^2-2\bar{X}X+\bar{X}^2)}}{N}
=\frac{\sum{X^2}=2\bar{X}\sum{X}+N\bar{X}^2}{N}
\end{align}<p>
\begin{align}
=\frac{\sum{X^2}}{N}-2\bar{X}\frac{\sum{X}}{N}+\bar{X}^2=\frac{\sum{X^2}}{N}-2\bar{X}^2+\bar{X}^2=\frac{\sum{X^2}}{N}-\bar{X}^2
\end{align}<p>
\begin{align}
=\overline{X^2}=\bar{X}^2=\frac{\sum{X^2}}{N}-\left(\frac{\sum{X}}{N}\right)2
\end{align}<p>
or
\begin{align}
s
=\sqrt{\frac{\sum{X^2}}{N}-\left(\frac{\sum{X}}{N}\right)^2}
=\sqrt{\overline{X^2}-\bar{X}^2}
\end{align}<p>
Note that in the above summations we have used the abbreviated form, with $X$ replacing $X_j$ and with $\sum$ replacing $\sum^N_{j=1}$.<p>
#### Another method
\begin{align}
s^2
=\overline{(X-\bar{X})^2}
=\overline{X^2-2X\bar{X}+\bar{X}^2}
=\overline{X^2}-\overline{2X\bar{X}}+\overline{\bar{X}^2}
=\overline{X^2}-2X\bar{X}+\bar{X}^2
=\overline{X^2}-\bar{X}^2
\end{align}<p>
(b)
\begin{align}
\overline{X^2}
=\frac{\sum{X^2}}{N}
=\frac{(12)^2+(6)^2+(7)^2+(3)^2+(15)^2+(10)^2+(18)^2+(5)^2}{8}
=\frac{912}{8}=114
\end{align}<p>
\begin{align}
\bar{X}
=\frac{\sum{X}}{N}
=\frac{12+6+7+3+15+10+18+5}{8}=\frac{76}{8}=9.5
\end{align}<p>
Thus
\begin{align}
s
=\sqrt{\overline{X^2}-\bar{X}^2}
=\sqrt{114-90.25}
=\sqrt{23.75}
=4.87
\end{align}<p>
This method should be compared with that of Problem 4.9(a).

In [23]:
p12.b <- c(12, 6, 7, 3, 15, 10, 18, 5)
p9.a.s
(p12.s <- round(sqrt(mean(p12.b^2) - mean(p12.b)^2),2))

[1] 4.87

[1] 4.87

---

__4.13__ Modify the formula of Problem 4.12(a) to allow for frequencies corresponding to the various values of $X$.

### Solution
The appropriate modification is<p>
\begin{align}
s
=\sqrt{\frac{\sum{fX^2}}{N}-\left(\frac{\sum{fX}}{N}\right)^2}
=\sqrt{\overline{X^2}-\bar{X}^2}
\end{align}<p>
As in Problem 4.12(a), this can be established by starting with<p>
\begin{align}
s
=\sqrt{\frac{\sum{f(X-\bar{X})^2}}{N}}
\end{align}<p>
Then
\begin{align}
s^2
=\frac{\sum{f(X-\bar{X})^2}}{N}
=\frac{\sum{f(X^2-2\bar{X}{X}+\bar{X}^2)}}{N}
=\frac{\sum{fX^2}-2\bar{X}\sum{fX}+\bar{X}^2\sum{f}}{N}
\end{align}<p>
\begin{align}
=\frac{\sum{fX^2}}{N}-2\bar{X}\frac{\sum{fX}}{N}+\bar{X}^2
=\frac{\sum{fX^2}}{N}-2\bar{X}^2+\bar{X}^2
=\frac{\sum{fX^2}}{N}-\bar{X}^2
\end{align}<p>
\begin{align}
=\frac{\sum{fX^2}}{N}-\left(\frac{\sum{fX}}{N}\right)^2
\end{align}<p>
or
\begin{align}
s=\sqrt{\frac{\sum{fX^2}}{N}-\left(\frac{\sum{fX}}{N}\right)^2}
\end{align}<p>
Note that in the above summations we have used the abbreviated form, with $X$ and $f$ replacing $X^j$ and $f^j$,$\sum$ replacing $\sum^K_{j=1}$, and $\sum^K_{j=1}f_j=N$.

---

__4.14__ Using the formula of Problem 4.13, find the standard deviation for the data in Table 4.2 of Problem 4.11.

### Solution
The work can be arranged as in the table, where $\bar{X}=\frac{\sum{fX}}{N}=67.45$in, as obtained in Problem 3.15. Note that this method, like that of Problem 4.11, entails much tedious computation. Problem 4.17 shows how the coding method simplifies the calculations immensely.

In [24]:
p14.X <- p11.table[,1]
p14.X.squared <- p14.X^2
p14.f <- p11.table[,4]
p14.f.X.squared <- p14.f*p14.X.squared
p14.table <- matrix( c(p14.X, p14.X.squared, p14.f, p14.f.X.squared), nrow=5)
rownames(p14.table) <- rownames(p11.table)
colnames(p14.table) <- c("Class Mark (X)", "X^2", "Frequency (f)", "fX^2")
p14.table

,Class Mark (X),X^2,Frequency (f),fX^2
60-62,61,3721,5,18605
63-65,64,4096,18,73728
66-68,67,4489,42,188538
69-71,70,4900,27,132300
72-74,73,5329,8,42632


\begin{align}
s
=\sqrt{\frac{\sum{fX^2}}{N}-\left(\frac{\sum{fX}}{N}\right)^2}
=\sqrt{\frac{455\text{,}803}{100}-(67.45)^2}=\sqrt{8.5275}=2.92\text{in}
\end{align}

In [25]:
p14.sum.f.X.squared <- sum(p14.f.X.squared)
p14.N <- sum(p14.f)
p14.X.bar <- sum(p14.f*p14.X)/p14.N
(p14.s <- round(sqrt(p14.sum.f.X.squared/p14.N-p14.X.bar^2),2))

[1] 2.92

---

__4.15__ If $d=X-A$ are the deviations of $X$ from an arbitrary constant $A$, prove that
\begin{align}
s
=\sqrt{\frac{\sum{fd^2}}{N}-\left(\frac{\sum{fd}}{N}\right)^2}
\end{align}

### Solution
Since $d=X-A, X=A,$ and $\bar{X}=A+\bar{d}$(see Problem 3.18), then<p>
\begin{align}
X-\bar{X}=(A+d)-(A+\bar{d})=d-\bar{d}
\end{align}<p>
so that
\begin{align}
s
=\sqrt{\frac{\sum{f(X-\bar{X})^2}}{N}}
=\sqrt{\frac{\sum{f(d-\bar{d})^2}}{N}}
=\sqrt{\frac{\sum{fd^2}}{N}-\left(\frac{\sum{fd}}{N}\right)^2}
\end{align}<p>
using the result of Problem 4.13 and replacing $X$ and $\bar{X}$ with $d$ and $\bar{d}$, respectively.
#### Another method
\begin{align}
s^2
=\overline{(X-\bar{X})^2}=\overline{(d-\bar{d})^2}=\overline{d^2-2\bar{d}d+\bar{d}^2}
\end{align}<p>
\begin{align}
=\overline{d^2}-2\bar{d}^2+\bar{d}^2=\overline{d^2}-\bar{d}^2=\frac{\sum{fd^2}}{N}-\left(\frac{\sum{fd}}{N}\right)^2
\end{align}<p>
and the result follows on taking the positive square root.

---

__4.16__ Show that if each class mark $X$ in a frequency distribution having class intervals of equal size $c$ is coded into a corresponding value $u$ according to the relation $X=A+cu$, where $A$ is a given class mark, then the standard deviation can be written as<p>
\begin{align}
s
=\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
=c\sqrt{\overline{u^2}-\bar{u}^2}
\end{align}

### Solution
This follows at once from Problem 4.15 since $d=X-A=cu$. Thus, since $c$ is a constant,<p>
\begin{align}
s
=\sqrt{\frac{\sum{f(cu)^2}}{N}-\left(\frac{\sum{f(cu)}}{N}\right)^2}
=\sqrt{c^2\frac{\sum{fu^2}}{N}-c^2\left(\frac{\sum{fu}}{N}\right)^2}
=c\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
\end{align}
#### Another method
We can also prove the result directly without using Problem 4.15. Since $X=A+cu, \bar{X}=A+c\bar{u},$ and $X-\bar{X}=c(u-\bar{u})$, then<p>
\begin{align}
s^2
=\overline{(X-\bar{X})^2}
=\overline{c^2(u-\bar{u})^2}
=c^2\overline{(u^2-2\bar{u}u+\bar{u}^2)}
=c^2(\overline{u^2}-2\bar{u}^2+\bar{u}^2)
=c^2(\overline{u^2}-\bar{u}^2)
\end{align}<p>
and
\begin{align}
s
=c\sqrt{\overline{u^2}-\bar{u}^2}
=c\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
\end{align}<p>

---

__4.17__ Find the standard deviation of the hieghts of the students at XYZ University (see Table 2.1) by using (a) the formula derived in Problem 4.15 and (b) the coding method of Problem 4.16.

### Solution
In Tables 4.4 and 4.5, $A$ is arbitrarily chosen as being equal to the class mark $67$. Note that in Table 4.4 the deviations $d=X-A$ are all multiples of the class-interval size $c=3$. This factor is removed in Table 4.5. As a result, the computations in Table 4.5 are greatly simplified (compare them with those of Problems 4.11 and 4.14). For this reason, the coding method should be used wherever possible.

(a) See Table 4.4 below

In [31]:
#load("Saves/Table_2.1.Rdata")
Table_4.4 <- matrix( c(61, 64, 67, 70, 73, -6, -3, 0, 3, 6, 5, 18, 42, 27, 8, -30, -54, 0, 81, 48, 180, 162, 0, 243, 288), nrow=5)
colnames(Table_4.4) <- c("Class Mark (X)", "d=X-A", "Frequency (f)", "fd", "fd^2")
rownames(Table_4.4) <- c("", "", "A", "", "")
Table_4.4
p17.a.N <- sum(Table_4.4[,3])
p17.a.fd.sum <- sum(Table_4.4[,4])
p17.a.fd.squared.sum <- sum(Table_4.4[,5])
(p17.a.s <- round(sqrt(p17.a.fd.squared.sum/p17.a.N-(p17.a.fd.sum/p17.a.N)^2),2))

,Class Mark (X),d=X-A,Frequency (f),fd,fd^2
,61,-6,5,-30,180
,64,-3,18,-54,162
A,67,0,42,0,0
,70,3,27,81,243
,73,6,8,48,288


[1] 2.92

\begin{align}
s
=\sqrt{\frac{\sum{fd^2}}{N}-\left(\frac{\sum{fd}}{N}\right)^2}
=\sqrt{\frac{873}{100}-\left(\frac{45}{100}\right)^2}
=\sqrt{8.5275}=2.92\text{in}
\end{align}

(b) see Table 4.5 below

In [30]:
Table_4.5 <- matrix(c(61,64,67,70,73, -2, -1, 0, 1, 2, 5, 18, 42, 27, 8, -10, -18, 0, 27, 16, 20, 18, 0, 27, 32), nrow=5)
colnames(Table_4.5) <- c("Class Mark (X)", "u=(X-A)/c", "Frequency (f)", "fu", "fu^2")
rownames(Table_4.5) <- c("", "", "A", "", "")
Table_4.5
p17.b.c <- 3
p17.b.N <- sum(Table_4.5[,3])
p17.b.fu.sum <- sum(Table_4.5[,4])
p17.b.fu.squared.sum <- sum(Table_4.5[,5])
(p17.b.s <- round(p17.b.c*sqrt(p17.b.fu.squared.sum/p17.b.N-(p17.b.fu.sum/p17.b.N)^2),2))

,Class Mark (X),u=(X-A)/c,Frequency (f),fu,fu^2
,61,-2,5,-10,20
,64,-1,18,-18,18
A,67,0,42,0,0
,70,1,27,27,27
,73,2,8,16,32


[1] 2.92

\begin{align}
s
=c\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
=3\sqrt{\frac{97}{100}-\left(\frac{15}{100}\right)^2}
=3\sqrt{0.9475}=2.92\text{in}
\end{align}

---

__4.18__ Using coding methods, find (a) the mean and (b) the standard deviation for the wage distribution of the $65$ employees at the P&R Company (see Table 2.5 of Problem 2.3).

### Solution
The work can be arranged simply, as shown in Table 4.6.
(a)
\begin{align}
\bar{X}=A+c\bar{u}=A+c\frac{\sum{fu}}{N}=\$275.00+(\$10.00)\left(\frac{31}{65}\right)=\$279.77
\end{align}
(b)
\begin{align}
s
=c\sqrt{\overline{u^2}-\bar{u}^2}
=c\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
=(\$10.00)\sqrt{\frac{173}{65}-\left(\frac{31}{65}\right)^2}
=(\$10.00)\sqrt{2.4341}=\$15.60
\end{align}

In [9]:
#load("Saves/Table_2.6.Rdata")
Table_4.6 <- matrix(c(255.00, 265.00, 275.00, 285.00, 295.00, 305.00, 315.00, -2, -1, 0, 1, 2, 3, 4, 8, 10, 16, 14, 10, 5, 2, -16, -10, 0, 14, 20, 15, 8, 32, 10, 0, 14, 40, 45, 32), nrow=7)
colnames(Table_4.6) <- c("X", "u", "f", "fu", "fu^2")
rownames(Table_4.6) <- c( "", "", "A", "", "", "", "")
Table_4.6
p18.N <- sum(Table_4.6[,3])
p18.fu.sum <- sum(Table_4.6[,4])
p18.fu.squared.sum <- sum(Table_4.6[,5])
p18.A <- Table_4.6[3,1]
p18.c <- 10
(p18.X.bar <- round(p18.A + p18.c*(p18.fu.sum/p18.N),2))
(p18.s <- round(p18.c*sqrt(p18.fu.squared.sum/p18.N-(p18.fu.sum/p18.N)^2),2))

,X,u,f,fu,fu^2
,255,-2,8,-16,32
,265,-1,10,-10,10
A,275,0,16,0,0
,285,1,14,14,14
,295,2,10,20,40
,305,3,5,15,45
,315,4,2,8,32


[1] 279.77

[1] 15.6

---

__4.19__ Table 4.7 shows the IQ's of 480 school children at a certain elementary school. Using the coding method, find (a) the mean and (b) the standard deviation.

In [19]:
p19.X <- c( 70, 74, 78, 82, 86, 90, 94, 98, 102, 106, 110, 114, 118, 122, 126)
p19.f <- c(4, 9, 16, 28, 45, 66, 85, 72, 54, 38, 27, 18, 11, 5, 2)
Table_4.7 <- t(matrix(c(p19.X, p19.f),ncol=2))
rownames(Table_4.7) <- c("Class mark (X)", "Frequency (f)")
Table_4.7

Class mark (X),70,74,78,82,86,90,94,98,102,106,110,114,118,122,126
Frequency (f),4,9,16,28,45,66,85,72,54,38,27,18,11,5,2


### Solution
The intelligence quotient is

\begin{align}
\text{IQ}=\frac{\text{mental age}}{\text{chronological age}}
\end{align}

expressed as a percentage. For example, an 8-year-old child who (according to certain educational procedures) has a mentality equivalent to that of a 10-year-old child would have an IQ of $10/8=1.25=125\%$, or simply $125$, the $\%$ sign being understood.

To find the mean and standard deviation of the IQ's in Table 4.7, we can arrange the work as in Table 4.8

(a)
\begin{align}
\bar{X}=A+c\bar{u}=A+c\frac{\sum{fu}}{N}=94+4\left(\frac{236}{480}\right)=95.97
\end{align}
(b)
\begin{align}
s
=c\sqrt{\overline{u^2}-u^2}
=c\sqrt{\frac{\sum{fu^2}}{N}-\left(\frac{\sum{fu}}{N}\right)^2}
=4\sqrt{\frac{3404}{480}-\left(\frac{236}{480}\right)^2}
=4\sqrt{6.8499}=10.47
\end{align}

In [26]:
p19.c <- 4
p19.A <- as.vector(Table_4.7[1,7])
p19.N <- sum(Table_4.7[2,])
p19.u <- c( -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8)
p19.fu <- p19.f*p19.u
p19.fu.sum <- sum(p19.fu)
(p19.X.bar <- round(p19.A + p19.c*(p19.fu.sum/p19.N),2))
p19.fu.squared.sum <- sum(p19.f*p19.u^2)
(p19.s <- round(p19.c*sqrt(p19.fu.squared.sum/p19.N-(p19.fu.sum/p19.N)^2),2))

[1] 95.97

[1] 10.47

---

## Charlier's check
Charlier's check in computations of the mean and standard deviation by the coding method makes use of the identities

\begin{align}
\sum{f(u+1)}=\sum{fu}+\sum{f}=\sum{fu}+N
\end{align}


\begin{align}
\sum{f(u+1)^2}=\sum{f(u^2+2u+1)}=\sum{fu^2}+2\sum{fu}+\sum{f}=\sum{fu^2}+2\sum{fu}+N
\end{align}

(See Problem 4.20)

---

__4.20__ Use Charlier's check to help verify the computations of (a) the mean and (b) the standard deviation performed in Problem 4.19.

### Solution
To supply the required check, the columns of Table 4.9 are added to those of Table 4.8 (with the exception of column 2, which is repeated in Table 4.9 for convenience).

(a) From Table 4.9, $\sum{f(u+1)}=716$; from Table 4.8, $\sum{fu}+N=236+480=716$. This provides the required check on the mean.

(b) From Table 4.9, $\sum{f(u+1)^2}=4356$; from Table 4.8, $\sum{fu^2}+2\sum{fu}+N=3404+2(236)+480=4356$. This provides the required check on the standard deviation.

In [34]:
p20.fu.squared <- p19.f*p19.u^2
p20.N <- p19.N
p20.fu.sum <- p19.fu.sum
p20.fu.squared.sum <- p19.fu.squared.sum
Table_4.8 <- matrix(c(p19.X, p19.u, p19.f, p19.fu, p20.fu.squared), nrow=15)
colnames(Table_4.8) <- c("X", "u", "f", "fu", "fu^2")
rownames(Table_4.8) <- c("", "", "", "", "", "", "A", "", "", "", "", "", "", "", "")
Table_4.8
Table_4.9 <- matrix(c(p19.u+1, p19.f, p19.f*(p19.u+1), p19.f*(p19.u+1)^2), nrow=15)
colnames(Table_4.9) <- c("u+1", "f", "f(u+1)", "f(u+1)^2")
(p20.N <- sum(p19.f))
(p20.fu1.sum <- sum(Table_4.9[,3]))
(p20.fu1.squared.sum <- sum(Table_4.9[,4]))
Table_4.9

,X,u,f,fu,fu^2
,70,-6,4,-24,144
,74,-5,9,-45,225
,78,-4,16,-64,256
,82,-3,28,-84,252
,86,-2,45,-90,180
,90,-1,66,-66,66
A,94,0,85,0,0
,98,1,72,72,72
,102,2,54,108,216
,106,3,38,114,342


[1] 480

[1] 716

[1] 4356

u+1,f,f(u+1),f(u+1)^2
-5,4,-20,100
-4,9,-36,144
-3,16,-48,144
-2,28,-56,112
-1,45,-45,45
0,66,0,0
1,85,85,85
2,72,144,288
3,54,162,486
4,38,152,608


---

## Properties of the standard deviation
1. The standard deviation can be defined as

\begin{align}
s=\sqrt{\frac{\sum^N_{j=1}{(X_j-a)^2}}{N}}
\end{align}

where $a$ is an average besides the arithmetic mean. Of all such standard deviations, the minimum is that for which $a=\bar{X}$, because of Property 2 in Chapter 3. This property provides an important reason for defining the standard deviation as above. For a proof of this property see Problem 4.27.